app/models/qing_group_fragment.rb

```diff
@@ -13,15 +13,22 @@ class QingGroupFragment
   attr_accessor :children, :qing_group
 
   delegate :hidden, :id, :group_name, :group_hint, :group_name_translations,
-    :group_hint_translations, :repeatable, to: :qing_group
+    :group_hint_translations, :repeatable, :odk_code, to: :qing_group
 
-  def initialize(qing_group)
+  def initialize(qing_group, children)
     self.qing_group = qing_group
-    self.children = ActiveSupport::OrderedHash.new
+    self.children = children
   end
 
   def multilevel?
-    children.keys.first.multilevel?
+    children.first.multilevel?
   end
 
+  def childless?
+    self.children.empty?
+  end
+
+  def sorted_children
+    self.children #already sorted in partitioner
```

Comment:
`self` is not needed in the above two cases and we don't include it by convention.